messagelib/messagelib.go

```diff
@@ -134,23 +134,44 @@ func getTestServerID() string {
 	return ""
 }
 
+// RemoveMentions removes any @ mentions from a message slice.
+func RemoveMentions(s []string) (ret []string) {
+	for _, v := range s {
+		if !strings.HasPrefix(v, "@") {
+			ret = append(ret, v)
+		}
+	}
+	return
+}
+
+// CanonicalizeMessage upcases each field in a message slice.
+func CanonicalizeMessage(s []string) (ret []string) {
+	for _, v := range s {
+		ret = append(ret, strings.ToUpper(v))
+	}
+	return
+}
+
 // ExpandAliases takes a string that contains an alias of format "?<alias>" and replaces the alias with the valid ticker string.
-func ExpandAliases(s string) string {
-	for k := range aliasMap {
-		s = strings.ReplaceAll(s, k, aliasMap[k])
+func ExpandAliases(s []string) (ret []string) {
+	for _, v := range s {
+		if strings.HasPrefix(v, "?") {
+			ret = append(ret, aliasMap[v]...)
+		} else {
+			ret = append(ret, v)
+		}
 	}
-	return s
+	return
 }
 
-// DedupeTickerStrings returns a list of unique tickers from the provided string slice.
-func DedupeTickerStrings(tickers []string) []string {
-	keys := make(map[string]bool)
-	list := []string{}
-	for _, entry := range tickers {
-		if _, value := keys[entry]; !value {
-			keys[entry] = true
-			list = append(list, entry)
+// DedupeSlice returns a list of unique tickers from the provided string slice.
+func DedupeSlice(s []string) (ret []string) {
```

Comment:
For my own edification: Is there really no golang library that does this? like converts a slice into what is a "Set"?

Reply:
There is no built-in to accomplish this.